src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Program.cs

```diff
@@ -119,6 +122,25 @@
         // Server has exited, cancel our service broker service
         await serviceBrokerFactory.ShutdownAndWaitForCompletionAsync();
     }
+
+    static void SetUICulture(ILogger logger, string? language)
+    {
+        if (language == null)
+        {
+            return;
+        }
+
+        try
+        {
+            var desiredUICulture = CultureInfo.CreateSpecificCulture(language);
+            CultureInfo.DefaultThreadCurrentUICulture = desiredUICulture;
+            CultureInfo.CurrentUICulture = desiredUICulture;
```

Comment:
Based on the docs I believe I need to set both - to ensure that the current thread culture is correct, and so that newly created threads use the correct culture.